### PR TITLE
Update target SDK to Android 14

### PR DESCRIPTION
**Pull request by irfano**
### Description

This PR makes the required change and updates the target SDK to 34 (Android 14).

> [!CAUTION]
> These steps should be completed before merging:
> - [x] Merge https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/pull/150
> - [x] Update the Login lib reference 
> - [x] Update the Play Core lib to a version compatible with Android 14. https://github.com/woocommerce/woocommerce-android/pull/11984

[Foreground service types are required](https://developer.android.com/about/versions/14/behavior-changes-14#fgs-types) now and this PR specifies the [short service](https://developer.android.com/about/versions/14/changes/fgs-types-required#short-service) type for `LoginWpcomService`.

`LoginWpcomService` is being used for the login process. If the user puts the app into the background while the login process is in progress, the service promotes itself to the foreground service and displays a notification like this:

<details>
  <summary> <a href="https://developer.android.com/develop/background-work/background-tasks#user-initiated">How to choose the right API for running a user-initiated background task.</a></summary>
 

</details>

According to the "How to choose the right API" chart, the short service is the best option for our case. because the login process can't be deferred or interrupted, and it's a short and critical task. If we don't add this service after targeting SDK 34, the app will crash.

### Managing timeout
The short service type has to end in about 3 minutes. "The app is expected to call `Service.stopSelf()` or `Service.stopForeground()` before the timeout occurs.". Logging in typically takes less than 3 minutes, but I have [handled onTimeout() in the Login Library](https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/pull/150), so we need to merge it first.
In this video, I manipulated the code and called timeout manually in 20 seconds. After `onTimeout()` is called, the notification shows a failure message.

[afterr.webm](https://github.com/woocommerce/woocommerce-android/assets/2471769/67cab5a4-c3f9-4631-bd85-dedf50cf7a71)

### Steps to reproduce

1. Launch the app on an Android 14 device.
2. Type your site name and email.
3. Type your password.
4. After typing your password and continuing, quickly navigate to the device's home screen.
5. Ensure the logging process is done after a while.

### Testing information

You may not always see the "Login in progress..." notification because OS may take up to 10 seconds to show a notification from the foreground.

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/225">19.5</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -19,6 +19,7 @@
         android:name="com.google.android.gms.permission.AD_ID"
         tools:node="remove" />
     <uses-permission android:name="android.permission.CAMERA" />
+    <uses-permission android:name="android.permission.FOREGROUND_SERVICE" />
```
Comment (hichamboushaba) [inline WooCommerce/src/main/AndroidManifest.xml]: We already declare the `FOREGROUND_SERVICE` above in line 12, we should remove this line.

**Review by hichamboushaba — APPROVED** [review]
Nice analysis and work @irfano 👏, ready to ship.

Note: since you are AFK, I committed a change to remove the duplicated permission declaration mentioned above.

File: WooCommerce/src/main/AndroidManifest.xml (hunk 1 above)
Comment (hichamboushaba) [reply WooCommerce/src/main/AndroidManifest.xml]: Removed in f12454c

Comment (irfano) [reply WooCommerce/src/main/AndroidManifest.xml]: Thank you!

**Pull request merged**